packages/block-library/src/query/hooks.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { createInterpolateElement } from '@wordpress/element';
+import { addQueryArgs } from '@wordpress/url';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { InspectorControls } from '@wordpress/block-editor';
+
+const CreateNewPostLink = ( {
+	attributes: { query: { postType } = {} } = {},
+} ) => {
+	if ( ! postType ) return null;
+	const newPostUrl = addQueryArgs( 'post-new.php', {
+		post_type: postType,
+	} );
+	return (
+		<div className="wp-block-query__create-new-link">
+			{ createInterpolateElement(
+				__( '<a>Create a new post</a> for this feed.' ),
+				// eslint-disable-next-line jsx-a11y/anchor-has-content
+				{ a: <a href={ newPostUrl } /> }
+			) }
+		</div>
+	);
+};
+
+/**
+ * Override the default edit UI to include layout controls
+ *
+ * @param  {Function} BlockEdit Original component
+ * @return {Function}           Wrapped component
+ */
+const queryTopInspectorControls = createHigherOrderComponent(
+	( BlockEdit ) => ( props ) => {
+		const { name } = props;
+		if ( name !== 'core/query' ) {
+			return <BlockEdit key="edit" { ...props } />;
+		}
+
+		return (
+			<>
+				<InspectorControls>
```

Comment:
Should probably be rendered conditionally? See #31381
We really need a better API here

Reply:
In this case is rendered conditionally when we edit `Query` block so I believe we are okay.

Reply:
I see a block name check, but not a selected check. Probably not significant for this block, but it will still render even if the block is not selected.

Reply:
I could also add the selected check as well.